libecole/src/observation/pseudocosts.cpp

```diff
@@ -0,0 +1,52 @@
+#include <cmath>
+#include <cstddef>
+
+#include <scip/scip_var.h>
+#include <scip/scipdefplugins.h>
+#include <scip/utils.hpp>
+
+#include "ecole/observation/pseudocosts.hpp"
+#include "ecole/scip/model.hpp"
+#include "ecole/scip/type.hpp"
+
+namespace ecole {
+namespace observation {
+
+nonstd::optional<xt::xtensor<double, 1>> Pseudocosts::obtain_observation(scip::Model& model) {
+
+	if (model.get_stage() == SCIP_STAGE_SOLVING) {
+
+		SCIP* scip = model.get_scip_ptr();
+
+		SCIP_VAR** cands;
+		SCIP_Real* cands_lp_values;
+		int nb_cands;
+
+		/* get branching candidates */
+		SCIPgetLPBranchCands(scip, &cands, &cands_lp_values, NULL, NULL, &nb_cands, NULL);
+		assert(nb_cands > 0);
+
+		/* Store pseudocosts in tensor */
+		auto const nb_lp_columns = static_cast<std::size_t>(SCIPgetNLPCols(scip));
+		auto pseudocosts = xt::xarray<double>::from_shape({nb_lp_columns});
+		pseudocosts.fill(std::nan(""));
+
+		SCIP_COL* col;
+		SCIP_Real score;
+		int lp_index;
```

Comment:
Variable in the loop don't need to be declared before the loop...

Reply:
I don't understand what you mean here.

Reply:
Wait, this is not resolved ^^
Let me do a commit for you to see

Reply:
Noooo! don't do that. I have the commit locally, now you just made a mess...